ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiLayeredWorldState.java

```diff
@@ -75,9 +75,29 @@ public Optional<BonsaiWorldView> getNextWorldView() {
   }
 
   public void setNextWorldView(final Optional<BonsaiWorldView> nextWorldView) {
+    maybeUnSubscribe();
```

Comment:
If we close the nextview, how will the layeredworldstate be able to ask it to retrieve the different values? Is it going to recreate it when needed?

Reply:
this is to close/unsubscribe from the worldstate this layer was initially created with, so we do not leak any snapshots.  since this is a mutator, this is the last opportunity we will have to unsubscribe, typically when the BonsaiWorldStateArchive is responding to a BlockAddedEvent, and changes the nextWorldView.